src/azureFunction/azureFunctionProjectService.ts

```diff
@@ -0,0 +1,167 @@
+			objectName,
+			constants.sqlConnectionString
+		);
+
+		this.refactorAzureFunction(functionFile);
+	}
+	/**
+	 * Refactors the Azure function file to include the sql binding specific function
+	 * @param filePath is the path for the function file (.cs for C# functions)
+	 */
+	private refactorAzureFunction(filePath: string): void {
+		let defaultBindedFunctionText = fs.readFileSync(filePath, 'utf-8');
+		// Replace default binding text
+		let newValueLines = defaultBindedFunctionText.split(os.EOL);
+		const defaultLineSet = new Set(defaultSqlBindingTextLines);
+		let replacedValueLines = [];
+		for (let defaultLine of newValueLines) {
+			// Skipped lines
+			if (defaultLineSet.has(defaultLine.trimStart())) {
+				continue;
+			} else if (defaultLine.trimStart() === defaultBindingResult) { // Result change
+				replacedValueLines.push(defaultLine.replace(defaultBindingResult, sqlBindingResult));
+			} else {
+				// Normal lines to be included
+				replacedValueLines.push(defaultLine);
+			}
+		}
+		defaultBindedFunctionText = replacedValueLines.join(os.EOL);
+		fs.writeFileSync(filePath, defaultBindedFunctionText, 'utf-8');
+	}
+}
+
+// A C# Azure Functions project must be present in order to create a new Azure Function for the table
+async function isAzureFunctionProjectOpen(): Promise<boolean> {
+	if (vscode.workspace.workspaceFolders === undefined || vscode.workspace.workspaceFolders.length === 0) {
+		return false;
+	} else {
+		const hostFile = await getHostFile();
+		const projectFiles = await getProjectFile();
+		if (projectFiles !== undefined && hostFile !== undefined) {
+			// select project to add azure function to
+			selectedProjectFile = (await vscode.window.showQuickPick(projectFiles, {
+				canPickMany: false,
+				title: constants.selectProject,
+				ignoreFocusOut: true
+			}));
+			return selectedProjectFile !== undefined;
+		}
+		return;
+	}
+}
+
+// gets the azure functions project file path
+async function getProjectFile(): Promise<string[] | undefined> {
+	const projUris = await vscode.workspace.findFiles('**/*.csproj');
+	let projFiles: string[] = [];
+	for (let projUri of projUris) {
+		projFiles.push(projUri.fsPath);
+	}
+	return projFiles.length > 0 ? projFiles : undefined;
+}
+
+// gets the host file config file path
+async function getHostFile(): Promise<string | undefined> {
+	const hostFiles = await vscode.workspace.findFiles('**/host.json');
+	return hostFiles.length > 0 ? hostFiles[0].fsPath : undefined;
+}
+
+// gets the local.settings.json file path
+async function getSettingsFile(): Promise<string | undefined> {
+	const settingsFiles = await vscode.workspace.findFiles('**/local.settings.json');
+	return settingsFiles.length > 0 ? settingsFiles[0].fsPath : undefined;
+}
+
+// retrieves the new function file once the file is created
+function getNewFunctionFile(): Promise<string> {
+	return new Promise((resolve) => {
+		const watcher = vscode.workspace.createFileSystemWatcher(new vscode.RelativePattern(
```

Comment:
We should really have a timeout here too, otherwise we might hang forever if something breaks creating the file.